src/libraries/System.Private.Uri/tests/FunctionalTests/UriTests.cs

```diff
@@ -819,23 +819,22 @@ public static void FilePathHandlesNonAscii(string uriString, string toString, st
             Assert.Equal(localPath, uri2.LocalPath);
         }
 
-        [Fact]
-        public static void ZeroPortIsParsedForBothKnownAndUnknownSchemes()
+        public static IEnumerable<object[]> ZeroPortIsParsedForBothKnownAndUnknownSchemes_TestData()
         {
-            Uri.TryCreate("http://example.com:0", UriKind.Absolute, out var uri);
-            Assert.Equal(0, uri.Port);
-            Assert.False(uri.IsDefaultPort);
-            Assert.Equal("http://example.com:0/", uri.ToString());
-
-            Uri.TryCreate("rtsp://example.com", UriKind.Absolute, out uri);
-            Assert.Equal(-1, uri.Port);
-            Assert.True(uri.IsDefaultPort);
-            Assert.Equal("rtsp://example.com/", uri.ToString());
+            yield return new object[] { "http://example.com:0", 0, false, "http://example.com:0/" };
```

Comment:
Is `toString` same as the `uriString` except the trailing slash? Do we need the extra parameter?